### PR TITLE
Upgrade TargetSdk, Kotlin, KSP, Gradle and Compose

### PR DESCRIPTION
**Pull request by pushpalroymm**
- Bump TargetSdk to `33` 
- Bump Kotlin to `1.8.10`
- Bump  KSP to `1.8.10-1.0.9`
- Bump  Gradle to `7.6`
- Bump AGP to `7.4.1`
- Bump Compose Compiler to `1.4.2`
- Add Compose BOM `2023.01.00`
- Replace deprecated code

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -1,16 +1,16 @@
 plugins {
     id 'com.android.application'
     id 'org.jetbrains.kotlin.android'
-    id 'com.google.devtools.ksp' version "1.6.10-1.0.4"
+    id 'com.google.devtools.ksp' version "1.7.21-1.0.8"
```
Comment (shubhamsinghmutualmobile) [inline app/build.gradle]: Can we go to the latest stable version of both Kotlin and KSP? For Kotlin, it's `1.8.10` and for KSP, it's `1.8.10-1.0.9`

File: build.gradle [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-        compose_version = '1.1.1'
+        compose_version = '1.4.0-alpha02'
     }
 }// Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id 'com.android.application' version '7.1.2' apply false
-    id 'com.android.library' version '7.1.2' apply false
-    id 'org.jetbrains.kotlin.android' version '1.6.10' apply false
+    id 'com.android.application' version '7.4.0' apply false
+    id 'com.android.library' version '7.4.0' apply false
```
Comment (shubhamsinghmutualmobile) [inline build.gradle]: Latest version is `7.4.1`

File: build.gradle [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+    id 'org.jetbrains.kotlin.android' version '1.7.21' apply false
```
Comment (shubhamsinghmutualmobile) [inline build.gradle]: 1.8.10

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#Wed Mar 09 11:42:03 IST 2022
+#Thu Feb 16 04:11:02 IST 2023
 distributionBase=GRADLE_USER_HOME
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-bin.zip
```
Comment (shubhamsinghmutualmobile) [inline gradle/wrapper/gradle-wrapper.properties]: Latest stable version is 7.6

File: settings.gradle [hunk 5]
```diff
@@ -1,8 +1,8 @@
 pluginManagement {
     plugins {
-        id("com.google.devtools.ksp") version "1.6.10-1.0.4"
+        id("com.google.devtools.ksp") version "1.7.21-1.0.8"
```
Comment (shubhamsinghmutualmobile) [inline settings.gradle]: 1.8.10-1.0.9

File: settings.gradle [hunk 6]
```diff
@@ -1,8 +1,8 @@
...
         id 'org.jetbrains.kotlin.jvm'
-        id 'org.jetbrains.kotlin.android' version '1.5.21'
+        id 'org.jetbrains.kotlin.android' version '1.7.21'
```
Comment (shubhamsinghmutualmobile) [inline settings.gradle]: 1.8.10

**Review by shubhamsinghmutualmobile — CHANGES_REQUESTED** [review]
Let's remove files coming in from `.idea`? And also, would it be a good time to introduce either `buildSrc` or composite builds to this project?

**Comment by pushpalroymm** [comment]
> Let's remove files coming in from `.idea`? And also, would it be a good time to introduce either `buildSrc` or composite builds to this project?

Thanks for pointing it out @shubhamsinghmutualmobile.
I have removed the cached files and added them in .gitignore.

Sure we can work on buildSrc/composite build in a separate PR. 👍

**Review by shubhamsinghmutualmobile — APPROVED** [review]
LGTM

**Review by ZeeshanShabbir — APPROVED** [review]

**Pull request merged**